deploy/lib/xquery/setup.xqy

```diff
@@ -54,6 +54,14 @@ declare variable $system-users := ("nobody", "infostudio-admin", "healthcheck");
 declare variable $system-roles as xs:string+ :=
   setup:read-config-file("security.xml")/sec:security/sec:roles/sec:role/@name;
 
+(: Return a set of messages to the user - add to the "messages" sequence :)
+declare variable $messages :=
+  map:map() !
+    (
+      map:put( ., "messages", () ),
+      .
+    );
+
```

Comment:
I'd probably use map:new and map:entry here, particularly since it uses the ! map operator. It works though. But apparently the same trick was used directly below before..

Reply:
I generally use the ! operator when creating a map, since it is much more performant than map:new and map:entry.  It obviously matters little when creating a small map, but for larger maps the performance can be significantly worse (especially when a module is being called repeatedly).

Reply:
Ah, hadn't considered performance, nor was I expecting so much difference. My consideration was simply that map:new/map:entry gives code that is easier to understand.

Reply:
Never mind. I might clean up in future. Or never..